### PR TITLE
[REM3-336] Fix the deadlock and update dependencies

### PR DESCRIPTION
**Pull request by fl4via**
Jira: https://issues.jboss.org/browse/REM3-336

### CODE REVIEW

File: src/main/java/org/jboss/remoting3/ConnectionPeerIdentityContext.java [hunk 1]
```diff
@@ -468,7 +468,7 @@ public ConnectionPeerIdentity getCurrentIdentity() {
 
         private final int id;
         private byte[] saslBytes;
-        private int status;
+        private volatile int status;
```
Comment (fl4via) [inline src/main/java/org/jboss/remoting3/ConnectionPeerIdentityContext.java]: @dmlloyd shouldn't this field be volatile? Maybe even a an AtomicFieldUpdater?

File: src/main/java/org/jboss/remoting3/EndpointImpl.java [hunk 2]
```diff
@@ -513,12 +513,17 @@ public void handleFailed(final IOException exception, final FutureResult<Connect
             }
 
             public void handleDone(final Connection connection, final FutureResult<ConnectionPeerIdentity> attachment) {
-                try {
-                    final ConnectionPeerIdentity identity = connection.getPeerIdentityContext().authenticate(authenticationConfiguration);
-                    futureResult.setResult(identity);
-                } catch (AuthenticationException e) {
-                    futureResult.setException(e);
-                }
+                worker.execute(() -> {
```
Comment (fl4via) [inline src/main/java/org/jboss/remoting3/EndpointImpl.java]: @dmlloyd use lambda even with the performance cost, or just create an anonymous Runnable?

Comment (dmlloyd) [reply src/main/java/org/jboss/remoting3/EndpointImpl.java]: I think either way is OK.

File: pom.xml [hunk 3]
```diff
@@ -44,15 +44,15 @@
...
+        <xnio.version>3.7.2.Final-SNAPSHOT</xnio.version>
         <logging.version>3.4.0.Final</logging.version>
         <logging-tools.version>2.2.0.Final</logging-tools.version>
         <logmanager.version>2.1.10.Final</logmanager.version>
         <wildfly-common.version>1.5.1.Final</wildfly-common.version>
         <modules.version>1.6.0.Final</modules.version>
         <junit.version>4.8.2</junit.version>
         <byteman-version>2.2.0.1</byteman-version>
-        <threads.version>2.2.1.Final</threads.version>
+        <threads.version>2.3.0.Beta2</threads.version>
```
Comment (dmlloyd) [inline pom.xml]: There is a 2.3.0.Final already, if you want to use that instead.

File: src/main/java/org/jboss/remoting3/ConnectionPeerIdentityContext.java (hunk 1 above)
Comment (dmlloyd) [reply src/main/java/org/jboss/remoting3/ConnectionPeerIdentityContext.java]: It's not necessary as all accesses to the field should already be protected by `synchronized` blocks.

**Pull request merged**